README.md

```diff
@@ -64,17 +64,30 @@ class MyDataSource(panoply.DataSource):
 
 The `DataSource` base class exposes the following methods:
 
-###### __init__(self, source, options)
+###### __init__(self, source, options, events)
 
-Constructor. Receives a dictionary with the data `source` parameters (see below) and a dictionary with any additional `options`. Generally, it's safe to disregard the options, however it may be used for performance optimizations, as it contains hints about incremental keys, excluded fields, etc.
+Constructor. Receives a dictionary with the data `source` parameters (see below), a dictionary with any additional `options` and a dictionary with `events` subscriptions. Generally, it's safe to disregard the options, however it may be used for performance optimizations, as it contains hints about incremental keys, excluded fields, etc, and it may contain additional parameters that can't be transferred with the source
```

Comment:
My immediate need was to change the source through the dynamic param function but why limit the SDK?
If you intend to give developers the ability to change the source (and you don't consider it "dangerous" via the collect process), why not give them the ability to change the source via the constructor or other methods? why do you consider this dangerous?

The API for dynamic function is still simple -> you get a list of arguments and you pass them on (e.g. `lambda *args, **kwargs: Source(*args, **kwargs).my_dynamic_func()`)

`DataSource` inherits `Emitter` which means that it already support "emitting" events, and the only "problem" is that there are no listeners.
Why do you think it's better to pass the `emit` function and not an event mapping? How would it make it less cryptic?